src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -198,4 +206,205 @@ mod YASPoolTests {
+            let tick_info = Tick::Info {
+                fee_growth_outside_0X128: 0,
+                fee_growth_outside_1X128: 0,
+                liquidity_gross: 1000,
+                liquidity_net: IntegerTrait::<i128>::new(0, false),
+                seconds_per_liquidity_outside_X128: 0,
+                tick_cumulative_outside: IntegerTrait::<i64>::new(0, false),
+                seconds_outside: 0,
+                initialized: true
+            };
+
+            // Init ticks
+            mock_ticks(ref tick_state, 0, 10, tick_info);
+
+            let delta_liquidity = IntegerTrait::<i128>::new(501, false);
+            InternalImpl::update_position(@pool_state, position_key, delta_liquidity, tick);
+        }
+    }
+
+    fn init_default(ref pool_state: ContractState) {
+        pool_state.set_fee(500);
+        pool_state.set_tick_spacing(IntegerTrait::<i32>::new(1, false));
+        pool_state.set_max_liquidity_per_tick(BoundedInt::max());
+        pool_state.set_fee_growth_globals(0, 0);
+    }
+
+    fn mock_contract_states() -> (
+        YASPool::ContractState, Position::ContractState, Tick::ContractState
+    ) {
+        let position_state = YASPool::InternalImpl::get_position_state(
+            YASPool::contract_state_for_testing()
+        );
+        let tick_state = YASPool::InternalImpl::get_tick_state(
+            YASPool::contract_state_for_testing()
+        );
+        (YASPool::contract_state_for_testing(), position_state, tick_state)
+    }
+
+    fn mock_position_key_and_info() -> (PositionKey, Position::Info) {
+        let position_key = PositionKey {
+            owner: OWNER(),
+            tick_lower: Zeroable::zero(),
+            tick_upper: IntegerTrait::<i32>::new(10, false),
+        };
+        let position_info = Info {
+            liquidity: 1000,
+            fee_growth_inside_0_last_X128: 0,
+            fee_growth_inside_1_last_X128: 0,
+            tokens_owed_0: 0,
+            tokens_owed_1: 0,
+        };
+        (position_key, position_info)
+    }
+
+    fn mock_tick_and_info() -> (i32, Tick::Info) {
+        let tick = IntegerTrait::<i32>::new(5, false);
+        let tick_info = Tick::Info {
+            fee_growth_outside_0X128: 0,
+            fee_growth_outside_1X128: 0,
+            liquidity_gross: 100,
+            liquidity_net: IntegerTrait::<i128>::new(0, false),
+            seconds_per_liquidity_outside_X128: 0,
+            tick_cumulative_outside: IntegerTrait::<i64>::new(0, false),
+            seconds_outside: 0,
+            initialized: true
+        };
+        (tick, tick_info)
+    }
+
+    fn mock_ticks(ref tick_state: Tick::ContractState, from: u32, to: u32, info: Tick::Info) {
+        assert(from <= to, 'init_mock_ticks - from > to');
+
+        let mut i = from;
+        loop {
+            if i == to + 1 {
+                break;
+            }
+            Tick::InternalImpl::set_tick(ref tick_state, IntegerTrait::<i32>::new(i, false), info);
```

Comment:
Another option would be to send a `Snap` and then do the iteration in `set_tick`

Reply:
Just added a method in the `InternalImpl` of `Tick`, which is `set_ticks()`, and it takes an array of ticks and infos. Check it out please